### PR TITLE
Test API logs rotation

### LINKED ISSUE #2636: Test API logs rotation
## Description

In #2580, a test to check the rotation of the `api.log` and `api.json` files was requested. But this test has a different approach than the other tests described in the issue mentioned above.

#### test_api_log_rotation.py
1. Given that the configured format is `json`, and the level is set to `info`, and the `wazuh-apid` is started; When the machine time is `midnight`; Then the JSON log file is rotated.
2. Given that the configured format is `plain`, and the level is set to `info`, and the `wazuh-apid` is started; When the machine time is `midnight`; Then the log file is rotated.
3. Given that the configured format is `json,plain`, and the level is set to `info`, and the `wazuh-apid` is started; When the machine time is `midnight`; Then both log files are rotated.
4. Given that the configured format is `plain,json`, and the level is set to `info`, and the `wazuh-apid` is started; When the machine time is `midnight`; Then both log files are rotated.

### PR DESCRIPTION
**Pull request by mauromalara**
|Related issue|
|---|
|#2636|

This PR closes #2636 

### Analysis
There is an API configuration option, called `logs`, which allows to log in 4 different ways ("json", "plain", "json,plain", and "plain,json") through the `format` field. When the API is configured with one of those values the logs are stored in the `api.log` and `api.json` files. When the machine time reaches **midnight** Wazuh rotates those files to `/var/ossec/logs/api/<YEAR>/<MONTH>`.

### Design
The test case design was done in #2580, but the tests were sliced due to they have different approaches.
#### test_api_log_rotation.py
The test aims to check that the API log files (api.log and api.json) are rotated properly, to achieve that we design the following test cases:

1. Given that the configured format is `json`, and the level is set to `info`, and the `wazuh-apid` is started; When the machine time is `midnight`; Then the JSON log file is rotated.
2. Given that the configured format is `plain`, and the level is set to `info`, and the `wazuh-apid` is started; When the machine time is `midnight`; Then the log file is rotated.
3. Given that the configured format is `json,plain`, and the level is set to `info`, and the `wazuh-apid` is started; When the machine time is `midnight`; Then both log files are rotated.
4. Given that the configured format is `plain,json`, and the level is set to `info`, and the `wazuh-apid` is started; When the machine time is `midnight`; Then both log files are rotated.

## Execution evidence
| Wazuh package | Package version |
|---|---|
| [package](https://packages-dev.wazuh.com/warehouse/test/4.4/rpm/var/wazuh-manager-4.4.0-api.logs.json.format.x86_64.rpm) | 4.4.0 |

<details>
<summary>Test execution - PASSED</summary>

```
python3 -m pytest tests/integration/test_api/test_config/test_logs/test_api_log_rotation.py
================================================================================================ test session starts =================================================================================================
platform linux -- Python 3.6.8, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /home/vagrant/wazuh-qa/tests/integration, configfile: pytest.ini
plugins: metadata-1.11.0, testinfra-5.0.0, html-3.1.1
collected 4 items                                                                                                                                                                                                    

tests/integration/test_api/test_config/test_logs/test_api_log_rotation.py ....                                                                                                                                 [100%]

=========================================================================================== 4 passed in 366.12s (0:06:06) ============================================================================================
```
</details>

## Tests

- [ ] Proven that tests pass when they have to pass.
- [ ] Proven that tests fail when they have to fail.
- [x] Python codebase satisfies PEP-8 style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] The test is documented following the QA-Docs schema.

### CODE REVIEW

**Comment by mauromalara** [comment]
## 08/03/2022
### Package
| Version | Revision | Link|
|---|---|---|
|4.4.0|api.logs.json.format|[package link](https://packages-dev.wazuh.com/warehouse/test/4.4/rpm/var/wazuh-manager-4.4.0-api.logs.json.format.x86_64.rpm)


## Testing

### API module

|  OS  | Local   | Jenkins     | Notes
|---    |---    |---    |---    |
| R1   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8209542/R1-2636-centos-manager.tar.gz)  |  SEE NOTE BELOW. | |
| R2   | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8209543/R2-2636-centos-manager.tar.gz)   |  SEE NOTE BELOW. | |
| R3   | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8209544/R3-2636-centos-manager.tar.gz)   |  SEE NOTE BELOW. | |

* Note: The feature development is blocked by a dependency that is not merged yet (see https://github.com/wazuh/wazuh/pull/11171 for further info). So, before executing the test it's necessary to install manually the dependency as follows: `/var/ossec/framework/python/bin/python3 -m pip install python-json-logger`

File: tests/integration/test_api/test_config/test_logs/test_api_log_rotation.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    current_formats = get_configuration['configuration']['logs']['format'].split(',')
+
+    yesterday = datetime.now() - timedelta(days=1)
+    year, month, day = yesterday.strftime(date_format_str_2).split('-')
+
+    get_api_details()
+    sleep(WAIT_FOR_MIDNIGHT)
+
+    if 'plain' in current_formats:
+        file_exists = os.path.isfile(os.path.join(WAZUH_PATH, 'logs', 'api', year, month, f'api.log-{day}.gz'))
```
Comment (Rebits) [inline tests/integration/test_api/test_config/test_logs/test_api_log_rotation.py:149]:
```suggestion
        file_exists = os.path.isfile(os.path.join(WAZUH_PATH, 'logs', 'api', year, month, f"api.log-{day}.gz"))
```

File: tests/integration/test_api/test_config/test_logs/test_api_log_rotation.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        assert file_exists, 'The plain log was not rotated.'
+    if 'json' in current_formats:
+        file_exists = os.path.isfile(os.path.join(WAZUH_PATH, 'logs', 'api', year, month, f'api.json-{day}.gz'))
```
Comment (Rebits) [inline tests/integration/test_api/test_config/test_logs/test_api_log_rotation.py:152]:
```suggestion
        file_exists = os.path.isfile(os.path.join(WAZUH_PATH, 'logs', 'api', year, month, f"api.json-{day}.gz"))
```

**Review by Rebits — CHANGES_REQUESTED** [review]
GJ, but some changes are required

**Comment by mauromalara** [comment]
## 14/03/2022
### Package
| Version | Revision | Link|
|---|---|---|
|4.4.0|api.logs.json.format|[package link](https://packages-dev.wazuh.com/warehouse/test/4.4/rpm/var/wazuh-manager-4.4.0-api.logs.json.format.x86_64.rpm)


## Testing

### API module

|  OS  | Local   | Jenkins     | Notes
|---    |---    |---    |---    |
| R1   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8245154/R1-2636-centos-manager.tar.gz)  |  SEE NOTE BELOW. | |
| R2   | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8245155/R2-2636-centos-manager.tar.gz)   |  SEE NOTE BELOW. | |
| R3   | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8245156/R3-2636-centos-manager.tar.gz)   |  SEE NOTE BELOW. | |

* Note: The feature development is blocked by a dependency that is not merged yet (see https://github.com/wazuh/wazuh/pull/11171 for further info). So, before executing the test it's necessary to install manually the dependency as follows: `/var/ossec/framework/python/bin/python3 -m pip install python-json-logger`

**Comment by Rebits** [comment]
## 14/03/2022
### Package
### Package
| Version | Revision | Link|
|---|---|---|
|4.4.0|api.logs.json.format|[package link](https://packages-dev.wazuh.com/warehouse/test/4.4/rpm/var/wazuh-manager-4.4.0-api.logs.json.format.x86_64.rpm)

## Testing

### `test_api`

|  OS  | Local | Notes
|---    |---    |---    |
| R1   | [:green_circle:  ](https://github.com/wazuh/wazuh-qa/files/8247057/R3-2646-API-logs.tar.gz)  |  |
| R2   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8247058/R2-2646-API-logs.tar.gz)  |  |
| R3   | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8247059/R1-2646-API-logs.tar.gz) |  |





* * * 

- :green_circle:: All pass
- :yellow_circle:: Some warnings
- :red_circle:: Some errors/fails
- :large_blue_circle:: In progress

**Pull request closed without merging**